app/lib/search_query_builder.rb

```diff
@@ -116,12 +122,28 @@ def default_order
     finder_content_item['details']['default_order'] || "-public_timestamp"
   end
 
-  def filter_query
-    filter_params
-      .merge(base_filter)
-      .reduce({}) { |query, (k, v)|
-        query.merge("filter_#{k}" => v)
-      }
+  def base_filter_query
+    @base_filter_query ||= base_filter.each_with_object({}) do |(k, v), query|
+      query["filter_#{k}"] = v
+    end
+  end
+
+  def and_filter_query
+    @and_filter_query ||= filter_params(combine_mode: 'and')
+      .each_with_object({}) do |(k, v), query|
+        query["filter_#{k}"] = v
+      end
+  end
+
+  def or_filter_queries
+    @or_filter_queries ||= filter_params(combine_mode: 'or')
+      .map do |k, v|
+        { "filter_#{k}" => v }
+      end
+  end
+
+  def filter_queries
+    (and_filter_query.empty? ? [] : [and_filter_query]) + or_filter_queries
```

Comment:
If the `and_filter_query` is put into the `base_filter_query`, I think `filter_queries` here can be dropped in favour of just using `or_filter_queries` directly.